### PR TITLE
Misc commits to android, json::dynamic and more_collections

### PR DESCRIPTION
**Pull request by xymus**
The gamepad events are based on 2 different gamepads. They will probably need to be revisited with different layouts.

### CODE REVIEW

File: lib/android/gamepad.nit [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	fun is_dpad_up: Bool do return key_code == 19
+
+	# Was `self` raised by the down key on the directional pad?
+	fun is_dpad_down: Bool do return key_code == 20
+
+	# Was `self` raised by the left key on the directional pad?
+	fun is_dpad_left: Bool do return key_code == 21
+
+	# Was `self` raised by the right key on the directional pad?
+	fun is_dpad_right: Bool do return key_code == 22
```
Comment (Morriar) [inline lib/android/gamepad.nit]: On some controllers you can also click on the stick.

File: lib/android/gamepad.nit [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+	# Was `self` raised by the start button?
+	fun is_start: Bool do return key_code == 108
+
+	# Was `self` raised by the select button?
+	fun is_select: Bool do return key_code == 109
+
+	# Was `self` raised by the right bumper button?
+	fun is_bumper_right: Bool do return key_code == 103
```
Comment (Morriar) [inline lib/android/gamepad.nit]: If this the button right atop the trigger?

**Comment by Morriar** [comment]
+1

File: lib/android/gamepad.nit (hunk 2 above)
Comment (xymus) [inline lib/android/gamepad.nit]: Exactly!

**Comment by jpages** [comment]
+1

**Comment by R4PaSs** [comment]
+1

**Pull request merged**